backend/query_conversion.go

```diff
@@ -0,0 +1,20 @@
+package backend
+
+import "context"
+
+// QueryConversionHandler is an EXPERIMENTAL service that allows converting queries between versions
+
+type QueryConversionHandler interface {
+	// ConvertQuery is called to covert queries between different versions
+	ConvertQuery(ctx context.Context, req *QueryDataRequest) (*QueryConversionResponse, error)
+}
+
+type ConvertQueryFunc func(ctx context.Context, req *QueryDataRequest) (*QueryConversionResponse, error)
+
+type QueryConversionResponse struct {
+	// Converted queries. It should extend v0alpha1.Query
+	QueryRequest *QueryDataRequest `json:"query"`
```

Comment:
```suggestion
	// Converted into typed queries
	Queries []any
```

With a typed query, then this same function can be used to execute each query and to clean them up